gateway-service/src/main/java/org/zowe/apiml/gateway/security/config/NewSecurityConfiguration.java

```diff
@@ -0,0 +1,394 @@
+     * Query endpoint does not require certificate to be present in RequestContext
+     */
+    @Configuration
+    @RequiredArgsConstructor
+    @Order(2)
+    class Query extends WebSecurityConfigurerAdapter {
+
+        private final TokenAuthenticationProvider tokenAuthenticationProvider;
+
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) {
+            auth.authenticationProvider(tokenAuthenticationProvider);
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            baseConfigure(http.requestMatchers().antMatchers(
+                authConfigurationProperties.getGatewayQueryEndpoint(),
+                authConfigurationProperties.getGatewayQueryEndpointOldFormat()
+                ).and()).authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .logout().disable() // logout filter in this chain not needed
+                .addFilterBefore(queryFilter("/**", authenticationManager()), UsernamePasswordAuthenticationFilter.class);
+        }
+
+        private QueryFilter queryFilter(String queryEndpoint, AuthenticationManager authenticationManager) {
+            return new QueryFilter(
+                queryEndpoint,
+                successfulQueryHandler,
+                handlerInitializer.getAuthenticationFailureHandler(),
+                authenticationService,
+                HttpMethod.GET,
+                false,
+                authenticationManager);
+        }
+    }
+
+    /**
+     * Query and Ticket endpoints share single filter that handles auth with and without certificate. This logic is encapsulated in the queryFilter or ticketFilter.
+     * Ticket endpoint does require certificate to be present in RequestContext
+     */
+
+    @Configuration
+    @RequiredArgsConstructor
+    @Order(3)
+    class Ticket extends WebSecurityConfigurerAdapter {
+
+        private final AuthenticationProvider tokenAuthenticationProvider;
+
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) {
+            auth.authenticationProvider(tokenAuthenticationProvider); // for authenticating Tokens
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            baseConfigure(http.requestMatchers().antMatchers(
+                authConfigurationProperties.getGatewayTicketEndpoint(),
+                authConfigurationProperties.getGatewayTicketEndpointOldFormat()
+            ).and()).authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .logout().disable() // logout filter in this chain not needed
+                .x509() //default x509 filter, authenticates trusted cert, ticketFilter(..) depends on this

```

Comment:
This looks like we could use our certificate to generate passticket. Do we read it properly?

Reply:
And is it actually valid use case that we should support?

Reply:
If we are calling through gateway, doesn't the certificate always comes from gateway for this case?

Reply:
there is ticket filter (actually query filter) on this endpoint. Ticket filter is authentication filter and issues the response and no routing happens here. Ticket filter verifies presence of Authentication object in security context holder, where it's put by the default `org.springframework.security.web.authentication.preauth.x509.X509AuthenticationFilter`, added to chain by `.x509()` method. I also tried to note that dependency with comment. Ticket filter also verifies the jwt token, so for calling this endpoint, you need both certificate and a jwt. As for support of functionality, there is no change in behavior on this endpoint.

Reply:
Does it mean that it's not possible to reach to this endpoint via the gateway routing? The way that we would reach out to other southbound services?